native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/internal/NativeImageCommandLineProvider.java

```diff
@@ -116,10 +124,68 @@ public Provider<RegularFile> getClasspathJar() {
     public List<String> asArguments() {
         NativeImageOptions options = getOptions().get();
         List<String> cliArgs = new ArrayList<>(20);
+        boolean hasLayers = !options.getLayers().isEmpty();
+        String layerCreateName = null;
+        ConfigurableFileCollection jarsClasspath = null;
+        if (hasLayers) {
+            LOGGER.warn("Experimental support for layered images enabled. DSL may change at any time.");
+            cliArgs.add("-H:+UnlockExperimentalVMOptions");
+            var layers = options.getLayers();
+            var arg = new StringBuilder();
+            for (LayerOptions layer : layers) {
+                if (arg.length() > 0) {
+                    arg.append(" ");
+                }
+                if (layer instanceof CreateLayerOptions) {
+                    var create = (CreateLayerOptions) layer;
+                    layerCreateName = layer.getLayerName().get();
+                    arg.append("-H:LayerCreate=");
+                    arg.append(layerCreateName).append(".nil");
+                    var modules = create.getModules().get();
+                    jarsClasspath = create.getJars();
+                    boolean hasModules = !modules.isEmpty();
+                    boolean hasPackage = create.getPackages().isPresent() && !create.getPackages().get().isEmpty();
+                    var createLayerJars = jarsClasspath.getFiles();
+                    boolean hasJars = !createLayerJars.isEmpty();
```

Comment:
Minor: maybe merge this line with the one above like:
`boolean hasJars = !jarsClasspath.getFiles().isEmpty();`
since `createLayerJars` is only used to calculate `hasJars` value.